test/test_layout.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <iostream>
+#include <numeric>
+
+#include "doctest/doctest.h"
+
+#include "sparrow/layout.hpp"
+
+namespace sparrow
+{
+    // TODO: Test all the other base types once #15 is addressed.
+    static_assert(std::ranges::range<primitive_layout<uint8_t>>);
```

Comment:
Nice! It just occured to me that in this specific case (primitive_layout) we can test it's a [contiguous_range](https://en.cppreference.com/w/cpp/ranges/contiguous_range) more specifically.